### PR TITLE
New module: `isoseq3/refine`

### PR DESCRIPTION
**Pull request by sguizard**
A new module for the program isoseq3 refine. 
It remove primers and concatemers from ccs sequences. 

## PR checklist

- [X] This comment contains a description of changes (with reason).
- [X] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [X] Remove all TODO statements.
- [X] Emit the `versions.yml` file.
- [X] Follow the naming conventions.
- [X] Follow the parameters requirements.
- [X] Follow the input/output options guidelines.
- [X] Add a resource `label`
- [X] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

**Comment by sguizard** [comment]
@nf-core/modules-team Hi, this module is ready for review.

File: modules/modtest/main.nf [hunk 1]
```diff
@@ -0,0 +1,82 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName; getProcessName } from './functions'
+
+// TODO nf-core: If in doubt look at other nf-core/modules to see how we are doing things! :)
```
Comment (drpatelh) [inline modules/modtest/main.nf:4]: Looks like you have accidentally tried to update 2 modules on this PR @sguizard ? The `modtest` one isn't finished? Could be worth creating separate PRs from scratch if it's easier.

Comment (sguizard) [reply modules/modtest/main.nf:4]: It's a dummy module I created to see the novelties of `versions.yaml`. 
I'm gonna delete this.

Comment (sguizard) [reply modules/modtest/main.nf]: Removed

File: modules/isoseq3/refine/main.nf [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    tuple val(meta), path(bam)
+    path(primers)
+
+    output:
+    tuple val(meta), path("*.flnc.bam")            , emit: bam
+    tuple val(meta), path("*.flnc.bam.pbi")        , emit: pbi
+    tuple val(meta), path("*.consensusreadset.xml"), emit: consensusreadset
+    tuple val(meta), path("*.filter_summary.json") , emit: summary
+    tuple val(meta), path("*.report.csv")          , emit: report
+    path  "versions.yml"                           , emit: version
```
Comment (drpatelh) [inline modules/isoseq3/refine/main.nf:31]:
```suggestion
    path  "versions.yml"                           , emit: versions
```

Comment (drpatelh) [inline modules/isoseq3/refine/main.nf:31]:
```suggestion
    tuple val(meta), path("*.bam")                 , emit: bam
    tuple val(meta), path("*.bam.pbi")             , emit: pbi
    tuple val(meta), path("*.consensusreadset.xml"), emit: consensusreadset
    tuple val(meta), path("*.filter_summary.json") , emit: summary
    tuple val(meta), path("*.report.csv")          , emit: report
    path  "versions.yml"                           , emit: versions
```

File: modules/isoseq3/refine/main.nf [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+    script:
+    def prefix   = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
```
Comment (drpatelh) [inline modules/isoseq3/refine/main.nf:34]:
```suggestion
    def prefix = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
```

File: modules/isoseq3/refine/main.nf [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    """
+    isoseq3 \\
+        refine \\
+        -j $task.cpus \\
+        $options.args \\
+        $bam \\
+        $primers \\
+        ${prefix}.flnc.bam
```
Comment (drpatelh) [inline modules/isoseq3/refine/main.nf:42]:
```suggestion
        ${prefix}.bam
```

File: modules/isoseq3/refine/main.nf [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+    cat <<-END_VERSIONS > versions.yml
+    ${getProcessName(task.process)}:
+        isoseq3 refine: \$( isoseq3 refine --version|sed 's/isoseq refine //'|sed 's/ (commit.\\+//' )
```
Comment (drpatelh) [inline modules/isoseq3/refine/main.nf:46]:
```suggestion
        ${getSoftwareName(task.process)}: \$( isoseq3 refine --version|sed 's/isoseq refine //'|sed 's/ (commit.\\+//' )
```

File: tests/modules/isoseq3/refine/main.nf [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+nextflow.enable.dsl = 2
+
+include { ISOSEQ3_REFINE } from '../../../../modules/isoseq3/refine/main' addParams( options: [:] )
+
+workflow test_isoseq3_refine {
+
+    input = [
+            [ id:'test' ], // meta map
+            file(params.test_data['homo_sapiens']['pacbio']['lima'],    checkIfExists: true),
+        ]
```
Comment (drpatelh) [inline tests/modules/isoseq3/refine/main.nf:12]:
```suggestion
    input = [
            [ id:'test' ], // meta map
            file(params.test_data['homo_sapiens']['pacbio']['lima'], checkIfExists: true),
    ]
```

File: tests/modules/isoseq3/refine/main.nf [hunk 7]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env nextflow
+
+nextflow.enable.dsl = 2
+
+include { ISOSEQ3_REFINE } from '../../../../modules/isoseq3/refine/main' addParams( options: [:] )
```
Comment (drpatelh) [inline tests/modules/isoseq3/refine/main.nf:5]:
```suggestion
include { ISOSEQ3_REFINE } from '../../../../modules/isoseq3/refine/main' addParams( options: [suffix:'refine'] )
```

File: tests/modules/isoseq3/refine/test.yml [hunk 8]
```diff
@@ -0,0 +1,16 @@
+- name: isoseq3 refine test_isoseq3_refine
+  command: nextflow run tests/modules/isoseq3/refine -entry test_isoseq3_refine -c tests/config/nextflow.config
+  tags:
+    - isoseq3
+    - isoseq3/refine
+  files:
+    - path: output/isoseq3/test.flnc.bam
```
Comment (drpatelh) [inline tests/modules/isoseq3/refine/test.yml:7]:
```suggestion
    - path: output/isoseq3/test.refine.bam
```

File: tests/modules/isoseq3/refine/test.yml [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+      md5sum: af2ac62a28727743ff18f8b1eb7c4269
+    - path: output/isoseq3/test.flnc.bam.pbi
+      md5sum: 8097cad9e472f2f79de6de5fe3dcc822
+    - path: output/isoseq3/test.flnc.consensusreadset.xml
+      contains: [ 'pbds:ConsensusReadSet' ]
+    - path: output/isoseq3/test.flnc.filter_summary.json
+      md5sum: 87f8bdd5c60741f47b8a991e002f7ef3
+    - path: output/isoseq3/test.flnc.report.csv
+      md5sum: d42a139e5d9b08396bdb087c01243ea9
```
Comment (drpatelh) [inline tests/modules/isoseq3/refine/test.yml:16]:
```suggestion
    - path: output/isoseq3/test.refine.bam
      md5sum: af2ac62a28727743ff18f8b1eb7c4269
    - path: output/isoseq3/test.refine.bam.pbi
      md5sum: 8097cad9e472f2f79de6de5fe3dcc822
    - path: output/isoseq3/test.refine.consensusreadset.xml
      contains: [ 'pbds:ConsensusReadSet' ]
    - path: output/isoseq3/test.refine.filter_summary.json
      md5sum: 87f8bdd5c60741f47b8a991e002f7ef3
    - path: output/isoseq3/test.refine.report.csv
      md5sum: d42a139e5d9b08396bdb087c01243ea9
```

File: tests/modules/isoseq3/refine/main.nf [hunk 10]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env nextflow
+
+nextflow.enable.dsl = 2
+
+include { ISOSEQ3_REFINE } from '../../../../modules/isoseq3/refine/main' addParams( options: [suffix:'refine'] )
```
Comment (drpatelh) [inline tests/modules/isoseq3/refine/main.nf:5]:
```suggestion
include { ISOSEQ3_REFINE } from '../../../../modules/isoseq3/refine/main' addParams( options: [suffix:'.refine'] )
```

File: tests/modules/isoseq3/refine/test.yml [hunk 11]
```diff
@@ -0,0 +1,16 @@
+- name: isoseq3 refine test_isoseq3_refine
+  command: nextflow run tests/modules/isoseq3/refine -entry test_isoseq3_refine -c tests/config/nextflow.config
+  tags:
+    - isoseq3
+    - isoseq3/refine
+  files:
+    - path: output/isoseq3/test.refine.bam
+      md5sum: af2ac62a28727743ff18f8b1eb7c4269
```
Comment (drpatelh) [inline tests/modules/isoseq3/refine/test.yml:8]:
```suggestion
      md5sum: e8387afd5f66a7f6a89f90a0dcf3b823
```

**Review by drpatelh — APPROVED** [review]
💯
